### PR TITLE
KAFKA-15853: Move ClientQuotaManagerConfig outside of core

### PR DESCRIPTION
**Pull request by OmniaGM**
blocker for #15158 

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: server/src/main/java/org/apache/kafka/server/config/ClientQuotaManagerConfig.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+ */
+package org.apache.kafka.server.config;
+
+public class ClientQuotaManagerConfig {
+    // Always have 10 whole windows + 1 current window
+    public static final int DEFAULT_NUM_QUOTA_SAMPLES = 11;
+    public static final int DEFAULT_QUOTA_WINDOW_SIZE_SECONDS = 1;
+
+    /**
+     * Configuration settings for quota management
```
Comment (nizhikov) [inline server/src/main/java/org/apache/kafka/server/config/ClientQuotaManagerConfig.java:28]: Javadoc describes constructor.
Move it, please.

File: core/src/main/scala/kafka/server/KafkaConfig.scala [hunk 2]
```diff
@@ -213,14 +213,14 @@ object Defaults {
   val FetchMaxBytes = 55 * 1024 * 1024
 
   /** ********* Quota Configuration ***********/
-  val NumQuotaSamples: Int = ClientQuotaManagerConfig.DefaultNumQuotaSamples
```
Comment (nizhikov) [inline core/src/main/scala/kafka/server/KafkaConfig.scala:216]: Can we keep existing order of variables?

File: server/src/main/java/org/apache/kafka/server/config/ClientQuotaManagerConfig.java [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+        this.numQuotaSamples = numQuotaSamples;
+        this.quotaWindowSizeSeconds = quotaWindowSizeSeconds;
+    }
+
+    public ClientQuotaManagerConfig(int numQuotaSamples) {
+        this.numQuotaSamples = numQuotaSamples;
+        this.quotaWindowSizeSeconds = DEFAULT_QUOTA_WINDOW_SIZE_SECONDS;
+    }
+
+    public int getNumQuotaSamples() {
```
Comment (nizhikov) [inline server/src/main/java/org/apache/kafka/server/config/ClientQuotaManagerConfig.java]: Both getters not used. Let's remove them

Comment (OmniaGM) [reply server/src/main/java/org/apache/kafka/server/config/ClientQuotaManagerConfig.java]: removed

**Review by nizhikov — APPROVED** [review]
Thanks for the PR. LGTM!

**Comment by nizhikov** [comment]
@mimaison 
@ijuma 

This PR are ready for merge for me. Please, take a look.

File: checkstyle/import-control-core.xml [hunk 4]
```diff
@@ -49,6 +49,7 @@
     <allow pkg="kafka.server"/>
     <allow pkg="kafka.tools"/>
     <allow pkg="org.apache.kafka.clients"/>
+    <allow pkg="org.apache.kafka.server"/>
```
Comment (mimaison) [inline checkstyle/import-control-core.xml]: I don't think we need the changes in this file

**Review by mimaison — COMMENTED** [review]
Thanks for the PR. It seems checkstyleMain/checkstyleTest pass without the changes in import-control-core.xml, are there changes really needed?

File: checkstyle/import-control-core.xml (hunk 4 above)
Comment (OmniaGM) [reply checkstyle/import-control-core.xml]: you right wit is not needed I think I forgot it while am breaking the main PR

Comment (OmniaGM) [reply checkstyle/import-control-core.xml]: removed it now

**Comment by nizhikov** [comment]
Hello @OmniaGM 

Can you, please, omit force push to the PR's? :)
This breaks commit history and reviewers (like I am) don't understand what have changed with force push.

PR will be squashed to the single nice looking commit during the merge, anyway.

File: checkstyle/import-control-core.xml (hunk 4 above)
Comment (mimaison) [reply checkstyle/import-control-core.xml]: Thanks, I'll wait for the CI to run then I'll take another look.

**Comment by mimaison** [comment]
@OmniaGM This needs rebasing too. Thanks

**Comment by OmniaGM** [comment]
> Hello @OmniaGM
> 
> Can you, please, omit force push to the PR's? :) This breaks commit history and reviewers (like I am) don't understand what have changed with force push.
> 
> PR will be squashed to the single nice looking commit during the merge, anyway.

I do this usually with medium/big prs but as this one small and we have a chain of prs relays on each others for this works this makes it easier to me to cherry-picking kinda for me.

**Comment by OmniaGM** [comment]
> @OmniaGM This needs rebasing too. Thanks

done

File: checkstyle/import-control-core.xml [hunk 5]
```diff
@@ -82,6 +82,7 @@
     <allow pkg="kafka.log" />
     <allow pkg="kafka.cluster" />
     <allow pkg="kafka.server" />
+    <allow pkg="org.apache.kafka.server" />
```
Comment (mimaison) [inline checkstyle/import-control-core.xml]: It looks like checkstyle works without these changes.

File: core/src/main/scala/kafka/server/ClientQuotaManager.scala [hunk 6]
```diff
@@ -32,6 +32,7 @@ import org.apache.kafka.common.security.auth.KafkaPrincipal
 import org.apache.kafka.common.utils.{Sanitizer, Time}
 import org.apache.kafka.server.config.ConfigEntityName
 import org.apache.kafka.server.quota.{ClientQuotaCallback, ClientQuotaEntity, ClientQuotaType}
+import org.apache.kafka.server.config.ClientQuotaManagerConfig
```
Comment (mimaison) [inline core/src/main/scala/kafka/server/ClientQuotaManager.scala]: Can we put it with the other `org.apache.kafka.server.config` import above?

File: server/src/main/java/org/apache/kafka/server/config/ClientQuotaManagerConfig.java [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+    }
+
+    public ClientQuotaManagerConfig() {
+        this(DEFAULT_NUM_QUOTA_SAMPLES, DEFAULT_QUOTA_WINDOW_SIZE_SECONDS);
+    }
+
+    public ClientQuotaManagerConfig(int numQuotaSamples) {
+        this(numQuotaSamples, DEFAULT_QUOTA_WINDOW_SIZE_SECONDS);
+    }
+}
```
Comment (mimaison) [inline server/src/main/java/org/apache/kafka/server/config/ClientQuotaManagerConfig.java]: Nit: We usually have a trailing line at the end of file.

**Review by mimaison — COMMENTED** [review]
Thanks for the updates. I left a couple of minor suggestions.

File: checkstyle/import-control-core.xml (hunk 5 above)
Comment (OmniaGM) [reply checkstyle/import-control-core.xml]: Remove it

File: core/src/main/scala/kafka/server/ClientQuotaManager.scala (hunk 6 above)
Comment (OmniaGM) [reply core/src/main/scala/kafka/server/ClientQuotaManager.scala]: done

File: server/src/main/java/org/apache/kafka/server/config/ClientQuotaManagerConfig.java (hunk 7 above)
Comment (OmniaGM) [reply server/src/main/java/org/apache/kafka/server/config/ClientQuotaManagerConfig.java]: Added a line

File: checkstyle/import-control-core.xml [hunk 8]
```diff
@@ -94,6 +94,7 @@
   <subpackage name="test">
     <allow pkg="org.apache.kafka.controller"/>
     <allow pkg="org.apache.kafka.metadata"/>
+    <allow pkg="org.apache.kafka.server"/>
```
Comment (mimaison) [inline checkstyle/import-control-core.xml]: I think we can remove that line too

Comment (OmniaGM) [reply checkstyle/import-control-core.xml]: remove it

**Review by mimaison — APPROVED** [review]
LGTM, thanks!

**Pull request merged**